lib/parser.js

```diff
@@ -0,0 +1,113 @@
+// ported from github.com/docker/docker
+// https://github.com/docker/docker/blob/e95b6b51daed868094c7b66113381d5088e831b4/builder/dockerfile/parser/line_parsers.go
+
+var parser = module.exports = {
```

Comment:
I'm going to have to trust you on this one.  It would be easier to check if you used a tokenizer.